lib/seahorn/BmcPass.cc

```diff
@@ -397,27 +397,40 @@ class BmcPass : public llvm::ModulePass {
 
     Stats::stop("BMC");
 
-    if (res == solver::SolverResult::SAT)
+    if (res == SolverResult::SAT)
       outs() << "sat";
-    else if (res == solver::SolverResult::UNSAT)
+    else if (res == SolverResult::UNSAT)
       outs() << "unsat";
     else
       outs() << "unknown";
     outs() << "\n";
 
-    if (res == solver::SolverResult::SAT)
+    if (res == SolverResult::SAT)
       Stats::sset("Result", "FALSE");
-    else if (res == solver::SolverResult::UNSAT)
+    else if (res == SolverResult::UNSAT)
       Stats::sset("Result", "TRUE");
 
-    LOG(
-        "cex", if (res == solver::SolverResult::SAT) {
-          errs() << "Analyzed Function:\n" << F << "\n";
-          errs() << "Trace \n";
-          SolverBmcTrace trace(bmc.getTrace());
-          trace.print(errs());
-        });
-    // todo: support --cex mode (generate harness)
+    if (res == SolverResult::SAT) {
+      SolverBmcTraceTy trace(bmc.getTrace());
```

Comment:
this whole `if` should be inside the `LOG` macro. If `LOG` is disabled, none of this code should be executed.